packages/core/src/components/image-overlay/image-overlay.tsx

```diff
@@ -16,34 +17,88 @@ export class ImageOverlay implements ComponentInterface {
   @State()
   focused = false;
 
+  @State()
+  zoomable = false;
+
   buttonElement: HTMLButtonElement | undefined;
-  
+
   wrapperElement: HTMLDivElement | undefined;
 
   trap: FocusTrap | undefined;
 
   private mutationObserver?: MutationObserver;
 
+  private resizeObserver = new ResizeObserver(debounce(() => {
+    const imgElement = this.host.querySelector('img');
+
+    if (imgElement instanceof HTMLImageElement) {
+      this.isImageZoomable(imgElement);
+    }
+  }, 200));
+
   componentDidLoad() {
-    this.mutationObserver = new MutationObserver(() => forceUpdate(this.host));
+    this.mutationObserver = new MutationObserver((e) => {
+      forceUpdate(this.host);
+
+      if (e[0]?.type === 'childList') {
+        // <img> is gone or a new element.
+        this.initZoomableImage();
+      }
+    });
+
     this.mutationObserver.observe(this.host, {
       attributes: true,
-      subtree: true
+      subtree: true,
+      attributeFilter: ['src', 'alt'],
+      childList: true,
     });
+
+    this.initZoomableImage();
   }
 
   disconnectedCallback() {
     this.trap?.deactivate();
     this.mutationObserver?.disconnect();
+    this.resizeObserver.disconnect();
+  }
+
+  initZoomableImage(): void {
+    this.resizeObserver.disconnect();
+
+    const imgElement = this.host.querySelector('img');
+
+    if (!(imgElement instanceof HTMLImageElement)) {
+      return;
+    }
+
+    imgElement.onload = (event) => {
```

Comment:
hiermee overschrijven we mogelijk een afnemer die ook `onload =` heeft gezet. Dit kan beter met `addEventListener();`